File: app/src/main/java/com/by5388/sy95306v2/main/model/IMainModel.java
package com.by5388.sy95306v2.main.model;

import io.reactivex.Observable;

/**
 * @author by5388  on 2019/1/2.
 */
public interface IMainModel {
    /**
     * 当前更新的进度、百分比
     *
     * @return 数量
     */
    Observable<Integer> insertProgressBar();

    /**
     * 车站总数量
     *
     * @return 总数量
     */
    Observable<Integer> getStationCount();

    /**
     * 是否需要更新，对比本地的数据版本与服务器的数据库版本
     *
     * @return 是否需要更新
     */
    Observable<Boolean> isNeedUpdate();

    /**
     * 网络是否可用
     *
     * @return 网络是否可用
     */
    boolean getNetWordCanUse();

    /**
     * 清空数据库
     *
     * @return 0:清空完成
     */
    Observable<Integer> clearData();

    /**
     * 更新完成，更新版本号
     */
    void finishUpdate();
}

File: sy95306v1kotlin/src/main/java/com/by5388/sy95306/kotlin/main/model/IMainModel.kt
package com.by5388.sy95306.kotlin.main.model

import io.reactivex.Observable

/**
 * @author  by5388  on 2019/1/8.
 */
interface IMainModel {
    /**
     * 检查是否需要更新
     * -1:时 不需要更新
     * 否则更新
     */
    fun checkUpdate(): Observable<Boolean>

    /**
     * 已经更新的数量
     */
    fun insertStationList(): Observable<Int>

    /**
     * 检查网络状态
     */
    fun getNetWordStatus(): Observable<Int>

    /**
     * 车站总数量
     *
     * @return 总数量
     */
    fun getStationCount(): Observable<Int>

    /**
     * 清空数据库
     *
     * @return 0:清空完成
     */
    fun clearData(): Observable<Int>

    /**
     * 更新完成，更新版本号
     */
    fun finishUpdate()
}